src/main/java/de/caritas/cob/mailservice/api/service/TranslationService.java

```diff
@@ -76,30 +74,41 @@ public Optional<Map<String, String>> tryFetchTranslations(String languageCode, D
       var result = fetchTranslationAsMap(languageCode, dialect);
       return result.isEmpty() ? Optional.empty() : Optional.of(result);
     } catch (JsonProcessingException e) {
-      log.warn("Error while processing json file with translations. Returning empty translations", e);
+      log.warn("Error while processing json file with translations. Returning empty translations",
+          e);
       return Optional.empty();
     }
   }
 
   private String fetchTranslationsAsString(String languageCode, Dialect dialect) {
     try {
-      return tranlationMangementServiceApiClient.tryFetchTranslationsFromTranslationManagementService(project, component,
+      return tranlationMangementServiceApiClient.tryFetchTranslationsFromTranslationManagementService(
+          project, component,
           languageCode, dialect);
     } catch (HttpClientErrorException e) {
       if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
-        log.warn("Translations for component {}, language {} not found in weblate, returning default translations", component,
+        log.warn(
+            "Translations for component {}, language {} not found in weblate, returning default translations",
+            component,
             languageCode);
-        return fetchDefaultTranslations(component, languageCode);
+        return fetchDefaultTranslations(component, languageCode, dialect);
       } else {
         log.error("Error while fetching translations from translation management service", e);
         throw e;
       }
+    } catch (ResourceAccessException ex) {
+      log.error("ResourceAccessException error while fetching translations from translation management service. Will fallback to resolve default translations.");
+      log.debug("Exception details: ", ex);
+      return fetchDefaultTranslations(component, languageCode, dialect);
     }
   }
 
-  private String fetchDefaultTranslations(String translationComponentName, String languageCode) {
+  private String fetchDefaultTranslations(String translationComponentName, String languageCode, Dialect dialect) {
+    String translationFilename = getTranslationFilename(
+        translationComponentName + "." + languageCode
+            + TranlationMangementServiceApiClient.getDialectSuffix(dialect));
```

Comment:
Shouldn't this fallback to mailservice.de.json? Or how does it work when its using dialect but we have no translation catalog for dialect stored in project?
In frontend there is a logic for fallback because not everything needs to be translated in dialect so we only dialect stuff in dialect catalog and all other translations are taken from the de catalog. Maybe we need a merge of them here too?

Reply:
hi @web-mi , do you want to have fallback logic to DE in case there is informal language setting in consultant profile but no informal file found? Ok, we can build this.
But I am against merging of json files. IMO if informal dialect can be set we should have all keys translated to dialect -> this should be required configuration. 
Otherwise some keys will have formal, another informal translations which will lead to a mix-up between Sie/Ihr and Du/Dir forms in the applications. 
So in worst case in one email user will get mixed up form. Either informal lang is fully supported and therefore all keys are translated, or not and in this case we don't need to have separate informal json file.

Reply:
Yes fallback logic if file does not exists locally. What is happening now when there is no informal file?

In Frontend we have 2700 lines inside de.json and only some of them have informal Translation. Thats why we have the fallback to de.
For example if we have a email subject "New message from advice seeker" its not required to translate it to informal because there is no formal language in this sentence.

Reply:
hey @web-mi , added the fallback logic in case informal translation file does not exist

Reply:
Oh dear, Marc has touched on an nice topic here. In the context of the mail service, I would be fine to have all translations duplicated and adapted where needed. That's how it would currently work, because it's a new component in Weblate.
However, with the many keys in the frontend, Marc's merge is of course important. Well, as I understand it, the frontend handles this itself, but of course it would be nice if all translation variants worked identically.
Do we need to have a chat about this?

Reply:
For frontend it makes sense to build a merge logic, but I really don't see added value for the mailservice. Frontend translations are huge and there are would be many repetitions with duplicated keys.
For mailservice we have small amount of labels, files are small and this will not change, therefore treating dialect as separate lang and having all keys translated there is not a problem. We avoid the need to have merge logic that has to be tested and maintained in future, increasing the complexity of the project.

Reply:
I was also speaking more from the perspective of the user operating Weblate... because they now need to understand how it works and where